File: M8/Exemplos/App_eXperience_Java/app/src/main/java/com/elgin/elginexperience/shipay/ShipayService.java
package com.elgin.elginexperience.shipay;

import com.elgin.elginexperience.shipay.requests.AuthenticationRequest;
import com.elgin.elginexperience.shipay.requests.CreateOrderRequest;
import com.elgin.elginexperience.shipay.responses.AuthenticationResponse;
import com.elgin.elginexperience.shipay.responses.CancelOrderResponse;
import com.elgin.elginexperience.shipay.responses.CreateOrderResponse;
import com.elgin.elginexperience.shipay.responses.GetOrderStatusResponse;
import com.elgin.elginexperience.shipay.responses.GetWalletsResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface ShipayService {
    @POST(Constants.AUTH_URL)
    Call<AuthenticationResponse> authenticate(@Body AuthenticationRequest request);

    @POST(Constants.CREATE_ORDER_URL)
    Call<CreateOrderResponse> createOrder(@Header("Authorization") String token, @Body CreateOrderRequest request);

    @GET(Constants.GET_WALLETS)
    Call<GetWalletsResponse> getWallets(@Header("Authorization") String token);

    @DELETE(Constants.CANCEL_ORDER)
    Call<CancelOrderResponse> cancelOrder(@Header("Authorization") String token, @Path("order_id") String orderId);

    @GET(Constants.GET_ORDER_STATUS)
    Call<GetOrderStatusResponse> getOrderStatus(@Header("Authorization") String token, @Path("order_id") String orderId);
}

File: Exemplos/App_eXperience_Kotlin/E1_KOTLIN_R/app/src/main/java/com/example/e1_kotlin_r/shipay/ShipayService.kt
package com.example.e1_elgin_kotlin.shipay

import com.example.e1_elgin_kotlin.shipay.requests.AuthenticationRequest
import com.example.e1_elgin_kotlin.shipay.requests.CreateOrderRequest
import com.example.e1_elgin_kotlin.shipay.responses.*
import retrofit2.Call
import retrofit2.http.*

interface ShipayService {
    @POST(Constants.AUTH_URL)
    fun authenticate(@Body request: AuthenticationRequest?): Call<AuthenticationResponse?>?

    @POST(Constants.CREATE_ORDER_URL)
    fun createOrder(
        @Header("Authorization") token: String?,
        @Body request: CreateOrderRequest?
    ): Call<CreateOrderResponse?>?

    @GET(Constants.GET_WALLETS)
    fun getWallets(@Header("Authorization") token: String?): Call<GetWalletsResponse?>?

    @DELETE(Constants.CANCEL_ORDER)
    fun cancelOrder(
        @Header("Authorization") token: String?,
        @Path("order_id") orderId: String?
    ): Call<CancelOrderResponse?>?

    @GET(Constants.GET_ORDER_STATUS)
    fun getOrderStatus(
        @Header("Authorization") token: String?,
        @Path("order_id") orderId: String?
    ): Call<GetOrderStatusResponse?>?
}